mtest: TestResult.SKIP is not a failure

smcv opened the pull request:
If some but not all tests in a run were skipped, then the overall result
is given by whether there were any failures among the non-skipped tests.

Resolves: https://github.com/mesonbuild/meson/issues/7515

jpakkane commented:
This needs a test. Can you get one of the original test failure logs, create a test executable that just prints that and then turn that into a test?

smcv commented:
> This needs a test. Can you get one of the original test failure logs, create a test executable that just prints that and then turn that into a test?

I added a minimal reproducer inline in `meson.build`, and a more realistic example using one of the GNOME tests that regressed like you said (the report of #7515 was for GTK, but the corresponding Debian bug was for gjs which compiles a lot more quickly, so I used a gjs test). Feel free to delete one or the other if you don't want both.

Instead of an executable that prints the output, I added a simple implementation of cat(1) and a text file with the output, so that it can be reused for other TAP sequences.

smcv commented:
Rebased on master to avoid CI failure (#7535).
